source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,182 @@
+      // over the surface.
+      double local_second_invariant_of_strain_rate_integral = 0.0;
+      double local_surface_area_integral = 0.0;
+
+      std::vector<Tensor<1,dim>> local_second_invariant_of_strain_rate(fe_face_values.n_quadrature_points);
+      std::vector<Tensor<1,dim>> local_surface_area(fe_face_values.n_quadrature_points); 
+      std::vector<std::pair<double,double>> local_second_invariant_of_strain_rate_and_corresponding_area; 
+      std::vector<SymmetricTensor<2,dim>> strain_rate (quadrature_formula.size());
+ 
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
+                {
+                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == top_boundary_id)
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            if (at_upper_surface)
+              {
+                fe_face_values.reinit(cell, face_idx);
+                fe_face_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(), strain_rate);
+              
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    //Merge so column 1 is strain rate invariants and column 2 are the corresponding areas
+                    local_second_invariant_of_strain_rate[q] = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
+                    local_surface_area[q] = fe_face_values.JxW(q);
+                    
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral = local_second_invariant_of_strain_rate[q] * local_surface_area[q];
+                    local_surface_area_integral += local_surface_area[q];
+                  }
+              }
+          }
+
+      const double total_second_invariant_of_strain_rate = Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator());
+      const double total_surface_area = Utilities::MPI::sum (local_surface_area_integral, this->get_mpi_communicator());
+
+      int n = sizeof(local_second_invariant_of_strain_rate)/sizeof(local_second_invariant_of_strain_rate[0]);
+      // Entering values in vector of pairs
+      for (int i=0; i<n; i++)
+        local_second_invariant_of_strain_rate_and_corresponding_area.push_back( std::make_pair(local_second_invariant_of_strain_rate[i],local_surface_area[i]));
```

Comment:
This cannot be a useful operation. The contents of `local_second_invariant_of_strain_rate[i]` and `local_surface_area[i]` are simply what you evaluated at the quadrature point of the last cell you visited. Which cell that is (or whether you visited any cell at all) is arbitrary, so I don't think that these lines do what you think they do. Can you explain in words what your goal is?

Reply:
In the "for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)" statement above this "for" loop I wanted to save the right value for each quadrature point into the following vectors "local_second_invariant_of_strain_rate" and "local_surface_area". So for the first quadrature point has it's values in the first row of the vectors above etc. Then I assumed that the "for (int i=0; i<n; i++)" statement you describe here- would just put these two vectors (local_second_invariant_of_strain_rate and local_surface_area) together side by side so it is a 2 column vector. This can then be sorted into smallest strains first and accumulated until the f_80 is reached. Is that not ok?